solrmonitor/SolrEventListener.go

```diff
@@ -0,0 +1,21 @@
+package solrmonitor
+
+/**
+The client can register the SolrEventListener to listen to the solr cluster events from the zookeeper.
+ */
+type SolrEventListener interface {
+	// all the live nodes in the solr cluster
+	SolrLiveNodesChanged(livenodes []string)
+
+	// all the query aggregator nodes in ths solr cluster
+	SolrQueryNodesChanged(querynodes []string)
+
+	// all the collections in the solr cluster
+	SolrCollectionsChanged(collections []string)
+
+	// current collection state change
+	SolrCollectionStateChanged(name string, collectionState *CollectionState)
+
+	// collection replica state changed
+	SolrCollectionReplicaStatesChanged(name string, replicaStates map[string]*PerReplicaState)
+}
```

Comment:
We might want to doc / provide links to Solr PRS right?  And be clear about how this works.  Like... what's the order of operations that the implementation is guaranteed to provide?

IE, it might be something like this:

On startup, the events are fired in this order:
- SolrCollectionsChanged fires for all collections that exist
- SolrCollectionStateChanged fires for each collection; this event does/does not include replica state for PRS collections?
- SolrCollectionReplicaStatesChanged for any PRS collections

When a collection is deleted:
- SolrCollectionStateChanged(nil) fires to notify the deletion
- SolrCollectionsChanged fires with an updated list of collections

When a collection is created:
- ...

When a collection state changes:
- ....

Get what I'm saying?  I have no idea if any of the words I wrote are accurate but hopefully you can fix :D

Reply:
Thanks, Scott. I have added this. Please let me know about this.